alert-common/src/main/java/com/synopsys/integration/alert/common/message/model/ComponentItem.java

```diff
@@ -114,6 +114,12 @@ public ComponentItem build() throws AlertException {
                 key = componentKey;
             } else {
                 final List<String> keyParts = new LinkedList<>(componentKeyParts);
+                keyParts.add(componentName);
+                keyParts.add(componentValue);
+                if (StringUtils.isNotBlank(subComponentName)) {
```

Comment:
We need to check name and value. I would argue value is the most important piece here.